numba_dpex/tests/test_parfors.py

```diff
@@ -0,0 +1,887 @@
+        self.state.typemap = None
+        self.state.return_type = None
+        self.state.calltypes = None
+        self.state.metadata = {}
+
+
+@skip_parfors_unsupported
+class TestParforBasic(TestParforsBase):
+    """Smoke tests for the parfors transforms. These tests check the most basic
+    functionality"""
+
+    def __init__(self, *args, **kwargs):
+        TestParforsBase.__init__(self, *args, **kwargs)
+        # these are used in the mass of simple tests
+        m = np.reshape(np.arange(12.), (3, 4))
+        self.simple_args = [np.arange(3.), np.arange(4.), m, m.T]
+
+    def test_simple01(self):
+        def test_impl():
+            return np.ones(())
+        with self.assertRaises(AssertionError) as raises:
+            self.check(test_impl)
+        self.assertIn("\'@do_scheduling\' not found", str(raises.exception))
+
+    def test_simple02(self):
+        def test_impl():
+            return np.ones((1,))
+        self.check(test_impl)
+
+    def test_simple03(self):
+        def test_impl():
+            return np.ones((1, 2))
+        self.check(test_impl)
+
+    def test_simple04(self):
+        def test_impl():
+            return np.ones(1)
+        self.check(test_impl)
+
+    def test_simple07(self):
+        def test_impl():
+            return np.ones((1, 2), dtype=np.complex128)
+        self.check(test_impl)
+
+    def test_simple08(self):
+        def test_impl():
+            return np.ones((1, 2)) + np.ones((1, 2))
+        self.check(test_impl)
+
+    def test_simple09(self):
+        def test_impl():
+            return np.ones((1, 1))
+        self.check(test_impl)
+
+    def test_simple10(self):
+        def test_impl():
+            return np.ones((0, 0))
+        self.check(test_impl)
+
+    def test_simple11(self):
+        def test_impl():
+            return np.ones((10, 10)) + 1.
+        self.check(test_impl)
+
+    def test_simple12(self):
+        def test_impl():
+            return np.ones((10, 10)) + np.complex128(1.)
+        self.check(test_impl)
+
+    def test_simple13(self):
+        def test_impl():
+            return np.complex128(1.)
+        with self.assertRaises(AssertionError) as raises:
+            self.check(test_impl)
+        self.assertIn("\'@do_scheduling\' not found", str(raises.exception))
+
+    def test_simple14(self):
+        def test_impl():
+            return np.ones((10, 10))[0::20]
+        self.check(test_impl)
+
+    def test_simple15(self):
+        def test_impl(v1, v2, m1, m2):
+            return v1 + v1
+        self.check(test_impl, *self.simple_args)
+
+    def test_simple16(self):
+        def test_impl(v1, v2, m1, m2):
+            return m1 + m1
+        self.check(test_impl, *self.simple_args)
+
+    def test_simple17(self):
+        def test_impl(v1, v2, m1, m2):
+            return m2 + v1
+        self.check(test_impl, *self.simple_args)
+
+    # for blas and lapack functions, we need scipy
+    # TODO: update README.md to conda install scipy
```

Comment:
@diptorupd for lapack and blas, we need scipy, therefore, to run this test we need to install it.

Reply:
@chudur-budur For some of the blas and lapck functions, we support them via dpnp. We do not want to test using scipy, for us the support needs to be via dpnp library routines.